lib/sisimai/bite/email/exchange2007.rb

```diff
@@ -0,0 +1,178 @@
+        connvalues = 0      # (Integer) Flag, 1 if all the value of $connheader have been set
+        connheader = {
+          'rhost' => '',    # The value of Reporting-MTA header or "Generating Server:"
+        }
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+
+            if connvalues == connheader.keys.size
+              # Diagnostic information for administrators:
+              #
+              # Generating server: mta2.neko.example.jp
+              #
+              # kijitora@example.jp
+              # #550 5.1.1 RESOLVER.ADR.RecipNotFound; not found ##
+              #
+              # Original message headers:
+              v = dscontents[-1]
+
+              if cv = e.match(/\A([^ @]+[@][^ @]+)\z/)
+                # kijitora@example.jp
+                if v['recipient']
+                  # There are multiple recipient addresses in the message body.
+                  dscontents << Sisimai::Bite.DELIVERYSTATUS
+                  v = dscontents[-1]
+                end
+                v['recipient'] = cv[1]
+                v['diagnosis'] = ''
+                recipients += 1
+
+              elsif cv = e.match(/\A[#]([45]\d{2})[ ]([45][.]\d[.]\d+)[ ].+\z/)
+                # #550 5.1.1 RESOLVER.ADR.RecipNotFound; not found ##
+                # #550 5.2.3 RESOLVER.RST.RecipSizeLimit; message too large for this recipien=
+                # t ##
+                v['replycode'] = cv[1]
+                v['status']    = cv[2]
+                v['diagnosis'] = e
+
+              else
+                if v['diagnosis'].to_s.size > 0 && v['diagnosis'] =~ /=\z/
+                  # Continued line of error messages
+                  v['diagnosis']  = v['diagnosis'].sub(/=\z/, '')
+                  v['diagnosis'] += e
+                end
+              end
+            else
+              # Diagnostic information for administrators:
+              #
+              # Generating server: mta22.neko.example.org
+              if cv = e.match(Re1[:rhost])
+                # Generating server: mta22.neko.example.org
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.
* Performance/EndWith: Use `String#end_with?` instead of a regex match anchored to the end of the string. [:ref](https://github.com/JuanitoFatas/fast-ruby#stringmatch-vs-stringstart_withstringend_with-code-start-code-end)

<!-- Comment by [SideCI](https://sideci.com) -->